pywr-schema/src/metric_sets/mod.rs

```diff
@@ -91,18 +131,46 @@ pub struct MetricSet {
     pub name: String,
     pub metrics: Vec<Metric>,
     pub aggregator: Option<MetricAggregator>,
+    #[serde(default)]
+    filters: MetricSetFilters,
 }
 
 impl MetricSet {
     #[cfg(feature = "core")]
     pub fn add_to_model(&self, network: &mut pywr_core::network::Network, args: &LoadArgs) -> Result<(), SchemaError> {
-        // Convert the schema representation to internal metrics.
-        let metrics: Vec<_> = self
+        use pywr_core::recorders::OutputMetric;
+
+        let mut metrics: Vec<OutputMetric> = self
             .metrics
             .iter()
+            //.chain(additional_metric.iter())
             .map(|m| m.load_as_output(network, args))
             .collect::<Result<_, _>>()?;
 
+        if let Some(additional_metrics) = self.filters.create_metrics(args) {
+            for m in additional_metrics.iter() {
```

Comment:
If `OutputMetric` derived `PartialEq` you can simplify this to:

```rust
            for m in additional_metrics.iter() {
                let o = m.load_as_output(network, args)?;
                if !metrics.contains(&o) {
                    metrics.push(o);
                }
            }
```

Reply:
Using a HashSet would be quicker, but requires `Hash` to be derived on `OutputMetric` and will also lose the ordering.